build.sbt

```diff
@@ -49,30 +50,46 @@ name := "cloudstate"
 
 val GrpcJavaVersion = "1.22.1"
 val GraalAkkaVersion = "0.5.0"
-val AkkaVersion = "2.5.31"
+val AkkaVersion = "2.6.5"
 val AkkaHttpVersion = "10.1.11"
 val AkkaManagementVersion = "1.0.5"
 val AkkaPersistenceCassandraVersion = "0.102"
 val PrometheusClientVersion = "0.6.0"
 val ScalaTestVersion = "3.0.5"
-val ProtobufVersion = "3.9.0"
+val ProtobufVersion = "3.9.0" // We use this version because it is the latest which works with native-image 20.0.0
 val GraalVersion = "20.0.0"
 val DockerBaseImageVersion = "adoptopenjdk/openjdk11:debian"
 val DockerBaseImageJavaLibraryPath = "${JAVA_HOME}/lib"
 
-def excludeTheseDependencies = Seq(
+val excludeTheseDependencies: Seq[ExclusionRule] = Seq(
   ExclusionRule("io.netty", "netty"), // grpc-java is using grpc-netty-shaded
   ExclusionRule("io.aeron"), // we're using Artery-TCP
   ExclusionRule("org.agrona") // and we don't need this either
 )
 
+def akkaDependency(name: String, excludeThese: ExclusionRule*) =
+  "com.typesafe.akka" %% name % AkkaVersion excludeAll ((excludeTheseDependencies ++ excludeThese): _*)
+
+def akkaHttpDependency(name: String, excludeThese: ExclusionRule*) =
+  "com.typesafe.akka" %% name % AkkaHttpVersion excludeAll ((excludeTheseDependencies ++ excludeThese): _*)
+
+def akkaManagementDependency(name: String, excludeThese: ExclusionRule*) =
+  "com.lightbend.akka.management" %% name % AkkaManagementVersion excludeAll ((excludeTheseDependencies ++ excludeThese): _*)
+
+def akkaDiscoveryDependency(name: String, excludeThese: ExclusionRule*) =
+  "com.lightbend.akka.discovery" %% name % AkkaManagementVersion excludeAll ((excludeTheseDependencies ++ excludeThese): _*)
+
+def akkaPersistenceCassandraDependency(name: String, excludeThese: ExclusionRule*) =
+  "com.typesafe.akka" %% name % AkkaPersistenceCassandraVersion excludeAll ((excludeTheseDependencies ++ excludeThese): _*)
+
 def common: Seq[Setting[_]] = Seq(
   headerMappings := headerMappings.value ++ Seq(
       de.heikoseeberger.sbtheader.FileType("proto") -> HeaderCommentStyle.cppStyleLineComment,
       de.heikoseeberger.sbtheader.FileType("js") -> HeaderCommentStyle.cStyleBlockComment
     ),
   // Akka gRPC overrides the default ScalaPB setting including the file base name, let's override it right back.
   akkaGrpcCodeGeneratorSettings := Seq(),
+  excludeDependencies += "com.typesafe.akka" %% "akka-protobuf-v3", // akka-protobuf-v3 shades protobuf-java 3.10.0 which is not compatible with native-image 20.0.0
```

Comment:
We need to exclude this across the board—this was the easiest way I could think of.

Reply:
How does it work if they are excluded? Won't excluding this break all the serializers that depend on it? Note, since the TCK just uses a cluster of one, ie it doesn't do any remoting, any breakage caused by excluding these probably would not be picked up by the TCK.

Reply:
Ah, I've now seen the comment below, disregard this.